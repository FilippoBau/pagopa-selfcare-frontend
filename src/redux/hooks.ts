import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux';
import {useHistory} from 'react-router-dom';
import type {RootState, AppDispatch} from './store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

<<<<<<< HEAD
export const useAppSelectorWithRedirect = ({
  selector,
  routeToRedirect,
  conditionToRedirect,
}: {
  selector: (state: RootState) => any;
  routeToRedirect?: string;
  conditionToRedirect?: (value: any) => boolean;
}) => {
  const selectedReduxState = useAppSelector(selector);
  const history = useHistory();
  const conditionResults = conditionToRedirect ? conditionToRedirect(selectedReduxState) : false;
  if (
    routeToRedirect &&
    (!selectedReduxState ||
      (typeof selectedReduxState === 'object' && Object.keys(selectedReduxState).length === 0) ||
      conditionResults)
  ) {
    history.push(routeToRedirect);
  }
  return selectedReduxState;
=======
export const useAppSelectorWithRedirect = (
    selector: (state: RootState) => any,
    routeToRedirect?: string
) => {
    const selectedReduxState = useAppSelector(selector);
    const history = useHistory();
    if (
        routeToRedirect &&
        (!selectedReduxState ||
            (typeof selectedReduxState === 'object' && Object.keys(selectedReduxState).length === 0))
    ) {
        history.push(routeToRedirect);
    }
    return selectedReduxState;
>>>>>>> 3f32cfc3 (Formatting (#542))
};
