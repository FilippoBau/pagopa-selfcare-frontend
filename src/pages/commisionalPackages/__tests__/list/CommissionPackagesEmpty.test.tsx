import { ThemeProvider } from '@mui/system';
import { theme } from '@pagopa/mui-italia';
import { cleanup, render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { store } from '../../../../redux/store';
import { Provider } from 'react-redux';
import React from 'react';
import CommissionPackagesEmpty from '../../list/CommissionPackagesEmpty';

//SNAPSHOT TESTING
it('renders correctly', () => {
  const tree = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={[`/comm-packages`]}>
        <Route path="/comm-packages">
          <ThemeProvider theme={theme}>
            <CommissionPackagesEmpty packageType={''} />
          </ThemeProvider>
        </Route>
      </MemoryRouter>
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});