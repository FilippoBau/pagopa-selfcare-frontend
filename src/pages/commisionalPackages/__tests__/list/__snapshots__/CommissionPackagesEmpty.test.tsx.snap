// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1fwt4j5"
      >
        <div
          class="MuiBox-root css-1podeb3"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-pjo5ov-MuiTypography-root"
          >
            Non sono ancora presenti pacchetti commissioni 
            
            .
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1lr2oxb-MuiTypography-root-MuiLink-root"
              href="/ui/comm-packages/add-package/"
            >
              <strong>
                 Crea Pacchetto
              </strong>
            </a>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1fwt4j5"
    >
      <div
        class="MuiBox-root css-1podeb3"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-pjo5ov-MuiTypography-root"
        >
          Non sono ancora presenti pacchetti commissioni 
          
          .
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1lr2oxb-MuiTypography-root-MuiLink-root"
            href="/ui/comm-packages/add-package/"
          >
            <strong>
               Crea Pacchetto
            </strong>
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
